Merge new Educators UX/UI in Open edX Docs

jswope00 opened the pull request:
As part of FC-0067, this is the revised Educators Homepage with UX based on behavioral navigation. Screenshot: 
![image](https://github.com/user-attachments/assets/56bcd21b-8048-4405-bdb3-33cb5c2d51bb)
![image](https://github.com/user-attachments/assets/7aa33b3e-a30b-415b-9d10-7f9640d84443)

openedx-webhooks commented:
Thanks for the pull request, @jswope00!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/docs-openedx-org-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

jswope00 commented:
Added commit [205ba08](https://github.com/openedx/docs.openedx.org/pull/601/commits/205ba0817a087798704e2af8eb095bceab47b8f3) to delete empty placeholder files and index.rst files that are now unecessary with the new UX. Added benefit is by removing those index.rst files, Sphinx will now catch any errors where files are not included in the new educators navigation.

sarina commented:
This looks great! I'm clicking around and I like what I see. It does have some conflicts, so those need to be resolved.

How would you like me to review this? I took a look at the diff and broadly it looks good - is there anything specific you'd like feedback on? I agree that removing the old index files is a good idea.

sarina commented on source/educators/navigation/advanced_features.rst:
```diff
@@ -0,0 +1,95 @@
+*******************************************************
+
+.. toctree::
+   :maxdepth: 2
```
I wonder if having maxdepth 2 on these pages makes the pages a bit more unwieldly.

https://docsopenedxorg--601.org.readthedocs.build/en/601/educators/navigation/advanced_features.html

jswope00 replied:
I'm also a little torn on that. Here is my argument for keeping it at 2: 

1. So far, we've broken these docs outs into their diataxis type but we still often have several how-to's in a single file  (e.g. Enable a Survey, Add a Survey, Edit a Survey) when they go together, and so we don't have 1,000+ files. If we did TOC level 1, we'd show something like this:
<img width="733" alt="image" src="https://github.com/user-attachments/assets/ff8f9697-cd64-4d3c-a589-41c1d33e03c7">

Instead of what we have now:
<img width="571" alt="image" src="https://github.com/user-attachments/assets/ce0f016a-7e52-47e6-b2e7-1fce0247d593">

OK, actually that's not bad. But I think it would involve renaming all the level 1 docs to make sure they action is clear (e.g. "Manage Survey Tool" instead of "Survey Tool"). This is something I'm open to do at the end of the project with leftover hours (I think there will be some). 

2. It makes the right sidebar navigation often redundant with the center panel. Especially for features where we only have one doc. e.g: 
<img width="434" alt="image" src="https://github.com/user-attachments/assets/a2db36db-193c-47dd-967c-f11f01e73877">

I say we bring this to the FC Group in our next meeting. But push this with 2 levels for now.

sarina replied:
OK, I agree with you. Leave at 2 levels for now, reasses after this merges.

sarina commented on source/educators/navigation/advanced_features.rst:
```diff
@@ -0,0 +1,95 @@
+   ../references/advanced_features/lti_canvas_example.rst
+   ../references/advanced_features/lti_blackboard_example.rst  
+
+Course Tags
```
```suggestion
Course Tags and Taxonomies
```

feanil commented on source/educators/index.rst:
```diff
@@ -1,76 +1,182 @@
+      :class-card: sd-shadow-md sd-p-2
+      :class-footer: sd-border-0
+
+      .. image:: /_images/educators_homepage/build_course_undraw.png
```
Right now when you click on the images, it opens a link to just the image.  Use the 'target' parameter to link the image to the same target as the `See All` link for all the cards.

jswope00 replied:
Fixed in this commit: [d63d26c](https://github.com/openedx/docs.openedx.org/pull/601/commits/d63d26c611119162a69cdf4130c2619ca9c237d9)

feanil reviewed (commented):
One suggestion otherwise looks good to me.

jswope00 commented:
> This looks great! I'm clicking around and I like what I see. It does have some conflicts, so those need to be resolved.
> 
> How would you like me to review this? I took a look at the diff and broadly it looks good - is there anything specific you'd like feedback on? I agree that removing the old index files is a good idea.

Where do you see conflicts? Mine shows "No conflicts, can be merged to main"?

sarina commented:
> Where do you see conflicts? Mine shows "No conflicts, can be merged to main"?

Hmm, maybe GH had a caching issue. Previously when I viewed the page, it said I couldn't merge due to conflicts.
